lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -542,19 +543,25 @@ newDBLayerWith _cacheBehavior tr ti SqliteContext{runQuery} = do
     -- assuming that the wallet already exists.
     let insertCheckpointCached wid cp = do
             liftIO $ traceWith tr $ MsgCheckpointCache wid MsgPutCheckpoint
-            modifyDBMaybe checkpointsVar $ \ws ->
-                let point = getPoint cp
-                in  case Map.lookup wid ws of
-                    Nothing  -> (Nothing, ())
-                    Just _   -> (Just $ Adjust wid $ PutCheckpoint point cp, ())
+            modifyDBMaybe walletsVar $ \ws ->
+                case Map.lookup wid ws of
+                    Nothing -> (Nothing, ())
+                    Just _  ->
+                        let (prologue, wcp) = fromWallet cp
+                            slot = getSlot wcp
+                            delta = Just $ Adjust wid
+                                [ UpdateCheckpoints $ PutCheckpoint slot wcp
```

Comment:
so basically this is a moment we do persistence, ie. store what is in `cp` and write to the DB, right? And thanks to construction of `Adjust` we will only write "difference" data, right? not all data from scratch?

Reply:
Essentially yes. More specifically, at this point in the code, we specify the delta that should be applied to the `Store`.

The actual update is performed by the `updateS` function of the `Store` here:

https://github.com/input-output-hk/cardano-wallet/blob/c649f5730632fa4498750dc6b111a43b380e42c2/lib/core/src/Cardano/Wallet/DB/Sqlite/CheckpointsOld.hs#L258-L259

Currently, this function still writes the entire `UTxO` and discovered addresses for the given checkpoint. In order to change that, we need to a) change the constructor `PutCheckpoint` to something like `PutDifferenceToPreviousCheckpoint` and b) change the database schema to accommodate this. One of the next few pull requests will do a), and one of the last commits in this series will be about b).

Reply:
ok, thanks for clarification